python/ray/data/tests/test_dataset.py

```diff
@@ -4233,7 +4233,7 @@ def f(should_import_polars):
         ctx.use_polars = original_use_polars
 
 
-def test_actorpoolstrategy_apply_interrupt():
+def test_actorpoolstrategy_apply_interrupt(shutdown_only):
```

Comment:
`shutdown_only` has to add here, o.w. the newly added unit test `test_actorpoolstrategy_default_max_actors` below cannot call `ray.init` again.

Reply:
Yep this is to be expected for tests that start a Ray cluster! If the cluster is shareable (e.g. same cluster config), the `ray_start_regular_shared` fixture takes care of sharing a common cluster across tests. But since this test needs a special cluster config, the `shutdown_only` fixture is the way to go.

Reply:
Supernit: `test_actor_pool_strategy_apply_interrupt` and `test_actor_pool_strategy_default_max_actors` may be a bit more readable, snake-casing the camel-cased class name instead of `actorpoolstrategy`. This is up to personal preference I think.

Reply:
@clarkzinzow - agree, updated.